### PR TITLE
SyncBot: Front events now possible in multiple inboxes.

### PR DESCRIPTION
**Pull request by sqweelygig**
**PLEASE NOTE** I have not been able to replicate the bug in my development environment due to not having aliased email accounts.  Two factors offset this however.  Firstly steps have been taken to reduce the occurrences of this bug in our production setup.  Secondly these changes are reasonably provable.

* Translators now create a list of messages from one event.
* Front translator expands a multiple inbox event into multiple messages.

Change-Type: patch
Connects-To: #217

---
##### Contributor checklist

- [ ] Build output been rebuilt and tested
- [ ] Introduces security considerations
- [ ] Tests are included
- [ ] Documentation is added or changed
- [ ] Affects the development, build or deployment processes of the component

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---
---- Autogenerated Waffleboard Connection: Connects to #217

### CODE REVIEW

File: lib/services/messenger.ts [hunk 1]
```diff
@@ -64,8 +64,10 @@ export class MessengerService extends ServiceScaffold<string> implements Service
 					listenerMethod: (_registration: ServiceRegistration, event: ServiceScaffoldEvent): Promise<void> => {
 						// This translates and enqueues any events that match a registered interest.
 						if (_.includes(this.eventsRegistered, subTranslator.eventIntoMessageType(event))) {
-							return subTranslator.eventIntoMessage(event)
-							.then(this.queueData);
+							return subTranslator.eventIntoMessages(event)
+							.then((messages) => {
+								_.forEach(messages, this.queueData);
```
Comment (brownjohnf) [inline lib/services/messenger.ts]: Does typescript automatically return the last line? Otherwise, this may not have completed before it's used.

Comment (hedss) [reply lib/services/messenger.ts]: This (correct me if I'm wrong Andrew) should end up queuing an event, which doesn't wait for a response but pushes the data into an async event queue. So, it doesn't actually need to wait for completion.

Comment (hedss) [reply lib/services/messenger.ts]: This (correct me if I'm wrong Andrew) should end up queuing an event, which doesn't wait for a response but pushes the data into an async event queue. So, it doesn't actually need to wait for completion.

Comment (sqweelygig) [reply lib/services/messenger.ts]: No correction needed, Hedss is entirely correct.

**Comment by sqweelygig** [comment]
I have built and tested it for single-inbox messages, and will do a paranoia check this before merging.

**Pull request merged**